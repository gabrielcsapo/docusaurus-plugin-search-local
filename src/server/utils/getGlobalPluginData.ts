import type { GlobalPluginData } from "docusaurus-plugin-search-local";
import type { PluginConfig } from "../../types";
import { getIndexHash } from "./getIndexHash";

export function getGlobalPluginData(
  pluginConfig: PluginConfig
): GlobalPluginData {
  const {
    externalSearchSources,
    removeDefaultStopWordFilter,
    searchResultContextMaxLength,
    searchResultLimits,
    translations,
    highlightSearchTermsOnTargetPage,
  } = pluginConfig;

  return {
    externalSearchSources,
    removeDefaultStopWordFilter,
    searchResultContextMaxLength,
    searchResultLimits,
    translations,
    highlightSearchTermsOnTargetPage,
    indexHash: getIndexHash(pluginConfig),
  };
}
