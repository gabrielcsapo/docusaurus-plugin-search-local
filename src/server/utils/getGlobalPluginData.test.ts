import fixturify from "fixturify";
import tmp from "tmp";

import type { PluginConfig } from "../../types";
import { getGlobalPluginData } from "./getGlobalPluginData";

describe("getGlobalPluginData", () => {
  const baselinePluginConfig: PluginConfig = {
    id: "foo",
    indexDocs: false,
    indexBlog: false,
    indexPages: false,
    docsRouteBasePath: ["docs"],
    blogRouteBasePath: ["blog"],
    hashed: false,
    docsDir: ["docs"],
    blogDir: ["blog"],
    removeDefaultStopWordFilter: false,
    highlightSearchTermsOnTargetPage: false,
    searchResultLimits: 8,
    searchResultContextMaxLength: 50,
    ignoreFiles: [],
    translations: {
      search_placeholder: "Search",
      see_all_results: "See all results",
      no_results: "No results.",
      search_results_for: 'Search results for "{{ keyword }}"',
      search_the_documentation: "Search the documentation",
      count_documents_found_plural: "{{ count }} documents found",
      count_documents_found: "{{ count }} document found",
      no_documents_were_found: "No documents were found",
    },
    externalSearchSources: [],
  };

  let tmpLocation: tmp.DirResult;
  let projectFixture;

  beforeEach(() => {
    tmpLocation = tmp.dirSync({ unsafeCleanup: true });
    projectFixture = {
      blog: {
        "hello-blog.md": `
          ---
          sidebar_position: 1
          ---
          
          # Hello Blog

          This is a dummy blog post.
        `,
      },
      docs: {
        "hello-world.md": `
          ---
          sidebar_position: 1
          ---
          
          # Hello World

          This is a dummy doc.
        `,
      },
    };

    fixturify.writeSync(tmpLocation.name, projectFixture);
  });

  afterEach(() => {
    tmpLocation.removeCallback();
  });

  test("it should generate global plugin data", () => {
    const config: PluginConfig = { ...baselinePluginConfig };

    expect(getGlobalPluginData(config)).toEqual({
      externalSearchSources: config.externalSearchSources,
      indexHash: null,
      highlightSearchTermsOnTargetPage: config.highlightSearchTermsOnTargetPage,
      removeDefaultStopWordFilter: config.removeDefaultStopWordFilter,
      searchResultContextMaxLength: config.searchResultContextMaxLength,
      searchResultLimits: config.searchResultLimits,
      translations: config.translations,
    });
  });

  test("it should generate an index hash when enabled", () => {
    const config: PluginConfig = {
      ...baselinePluginConfig,
      blogDir: [`${tmpLocation.name}/blog`],
      docsDir: [`${tmpLocation.name}/docs`],
      hashed: true,
      indexDocs: true,
      indexBlog: true,
    };

    expect(getGlobalPluginData(config)).toMatchSnapshot();
  });

  test("it should only index blog when enabled", () => {
    const config: PluginConfig = {
      ...baselinePluginConfig,
      blogDir: [`${tmpLocation.name}/blog`],
      hashed: true,
      indexBlog: true,
    };

    expect(getGlobalPluginData(config)).toMatchSnapshot();
  });

  test("it should only index docs when enabled", () => {
    const config: PluginConfig = {
      ...baselinePluginConfig,
      docsDir: [`${tmpLocation.name}/docs`],
      hashed: true,
      indexDocs: true,
    };

    expect(getGlobalPluginData(config)).toMatchSnapshot();
  });
});
