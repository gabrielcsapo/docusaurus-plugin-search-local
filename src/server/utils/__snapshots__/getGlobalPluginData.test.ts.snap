// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getGlobalPluginData > it should generate an index hash when enabled 1`] = `
{
  "externalSearchSources": [],
  "highlightSearchTermsOnTargetPage": false,
  "indexHash": "CONSISTENT_HASH",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for "{{ keyword }}"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;

exports[`getGlobalPluginData > it should only index blog when enabled 1`] = `
{
  "externalSearchSources": [],
  "highlightSearchTermsOnTargetPage": false,
  "indexHash": "CONSISTENT_HASH",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for "{{ keyword }}"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;

exports[`getGlobalPluginData > it should only index docs when enabled 1`] = `
{
  "externalSearchSources": [],
  "highlightSearchTermsOnTargetPage": false,
  "indexHash": "CONSISTENT_HASH",
  "removeDefaultStopWordFilter": false,
  "searchResultContextMaxLength": 50,
  "searchResultLimits": 8,
  "translations": {
    "count_documents_found": "{{ count }} document found",
    "count_documents_found_plural": "{{ count }} documents found",
    "no_documents_were_found": "No documents were found",
    "no_results": "No results.",
    "search_placeholder": "Search",
    "search_results_for": "Search results for "{{ keyword }}"",
    "search_the_documentation": "Search the documentation",
    "see_all_results": "See all results",
  },
}
`;
